openedx/core/lib/tests/test_xblock_utils.py

```diff
@@ -0,0 +1,195 @@
+
+    def test_replace_course_urls(self):
+        test_course = CourseKey.from_string('TestX/TS01/2015')
+        test_replace = replace_course_urls(
+            course_id=test_course,
+            block=CourseFactory.create(),
+            view='baseview',
+            frag=Fragment(u'<a href="/course/id">'),
+            context=None
+        )
+        self.assertIsInstance(test_replace, Fragment)
+        self.assertEqual(test_replace.content, u'<a href="/courses/TestX/TS01/2015/id">')
+
+    def test_replace_static_urls(self):
+        test_course = CourseKey.from_string('TestX/TS01/2015')
+        test_replace = replace_static_urls(
+            data_dir=None,
+            course_id=test_course,
+            block=CourseFactory.create(),
+            view='baseview',
+            frag=Fragment(u'<a href="/static/id">'),
+            context=None
+        )
+        self.assertIsInstance(test_replace, Fragment)
+        self.assertEqual(test_replace.content, u'<a href="/c4x/TestX/TS01/asset/id">')
+
+    def test_grade_histogram(self):
+        test_course = CourseKey.from_string('TestX/TS01/2015')
+        usage_key = test_course.make_usage_key('problem', 'first_problem')
+        new_grade = StudentModule.objects.create(
+            student_id=1,
+            grade=100,
+            module_state_key=usage_key
+        )
+        new_grade = StudentModule.objects.create(
+            student_id=2,
+            grade=50,
+            module_state_key=usage_key
+        )
+
+        grades = grade_histogram(usage_key)
+        self.assertEqual(grades[0], (50.0, 1))
+        self.assertEqual(grades[1], (100.0, 1))
+
+    def test_add_staff_markup(self):
+        block = CourseFactory.create()
+        block.location.block_id = 'test-test'
+        self.assertEqual(block.location.__dict__, 1)
+        output = add_staff_markup(
+            user=UserFactory(),
+            has_instructor_access=True,
+            disable_staff_debug_info=False,
+            block=CourseFactory.create(),
+            view='baseview',
+            frag=Fragment(u"<h1>Test!</h1>"),
+            context=None
+        )  # pylint: disable=unused-argument
+
+        self.assertEqual(output.__dict__, 1)
+
+    def test_get_course_update_items(self):
+        pass
+
+
+class AddStaffMarkup(ModuleStoreTestCase):
+
+    def setUp(self):
+        super(AddStaffMarkup, self).setUp(create_user=True)
+
+        self.content_store = contentstore()
+        # pylint: disable=protected-access
+        self.module_store = modulestore()._get_modulestore_by_type(ModuleStoreEnum.Type.split)
+
+    def test_add_staff_markup(self):
+        course = import_course_from_xml(
+            self.module_store, self.user.id, TEST_DATA_DIR,
+            ['problem_colon_id'],
+            create_if_not_present=True
+        )
+        course_key = course[0].id
+        problem = self.module_store.get_item(course_key.make_usage_key('problem', 'haveacolon'))
```

Comment:
Oh no, we love tests :). Out of curiousity, does this PR boost our coverage? That could be a good indicator of whether or not those functions were covered.

And re testing in modulestore, that makes sense. I'll hang back then until you're ready for a code review. But please do let me know if I can otherwise be helpful.